docs/10-lifecycle.md

```diff
@@ -0,0 +1,81 @@
+# Lifecycle
+
+ContainerPilot's core functionality is centered around the concepts of jobs, the events that trigger those jobs or that jobs emit, and watches for the external service discovery catalog (Consul).
```

Comment:
I agree with that sentiment. Users need to understand lifecycle as a concept first before they can understand jobs/events/etc.

Perhaps:

> ContainerPilot was created to manage the lifecycle of an application. It orchestrates the basic concept of application lifecycle – an object is created, things happen, the object is destroyed – the details differ based on all of the [jobs](#what-is-a-job) and  [events that trigger or are trigged by jobs](#what-is-an-event). All of this is then [watched](#what-is-a-watch) by the external service discovery catalog (Consul).

Each item linked to the relevant section beneath it.